cashu/wallet/api/error_handler.py

```diff
@@ -0,0 +1,17 @@
+from fastapi import HTTPException, Request, status
+from fastapi.routing import APIRoute
+
+
+class ErrorHandler(APIRoute):
+    def get_route_handler(self):
+        original_route_handler = super().get_route_handler()
+
+        async def custom_route_handler(request: Request):
+            try:
+                return await original_route_handler(request)
+            except Exception as exc:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST, detail=str(exc)
+                )
```

Comment:
Would suggest

```suggestion
            except (AssertionError, KeyboardInterrupt):
                # These should never really be caught, 1st is "something real bad" and 2nd is a PITA if something gets stuck
                raise
            except Exception as exc:
                if isinstance(exc, HTTPException):
                    # Allows optionally setting a custom `status_code` if desired, but otherwise defaulting to below for convenience
                    raise
                raise HTTPException(
                    status_code=status.HTTP_400_BAD_REQUEST, detail=str(exc)
                )
```

Reply:
`str(exc)` probably OK for now, but should be a little cautious about echoing the contents of whatever exception gets raised back to some random person on the internet, it can leak info about the system and make it easier to find vulnerabilities (minor example #228 where the API responses reveal that this mint was running Postgres and CLN specifically)

Reply:
Would still recommend passing-up a `HTTPException` as-is if one is encountered, but could easilly be added later if someone needs this

```python
if isinstance(exc, HTTPException):
    # Allows optionally setting a custom `status_code` if desired, but otherwise defaulting to below for convenience
    raise
```

---

On the `AssertionError`, seems `assert` has been used in a few places so can pass on that suggestion (though IMHO even though `assert` is shorter, it is best reserved as a thing to actually cause a crash, like if somehow `1 != 1`)